docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,64 @@
+
+- [Etcher](https://github.com/resin-io/etcher)
+
+This is the *"main repository"*, from which you're reading this from, which is basically the front-end and glue for all previously listed projects.
+
+Front-ends
+----------
+
+The main repository consists of the implementation of the Etcher CLI and the Etcher GUI (the desktop application), located at [`lib/cli/`](https://github.com/resin-io/etcher/tree/master/lib/cli) and [`lib/gui/`](https://github.com/resin-io/etcher/tree/master/lib/gui), respectively.
+
+In fact, the only front-end that interacts directly with Etcher's backend is the CLI. The GUI merely forks the CLI and communicates with its child process to get state information. 
+
+In this sense, you can consider the GUI as being the front-end to the CLI, which is in turn the front-end to the actual image writing functionality.
+
+As a way to simplify how the GUI forks the CLI in a packaged and distributed context, both the CLI and GUI share the same application entry point. This means that the same Etcher binary can behave as CLI or GUI as needed.
+
+## Process communication
+
+As mentioned before, the Etcher GUI forks the CLI and retrieves information from it to update its state. In order to accomplish this, the Etcher CLI contains certain features to ease communication:
+
+- [Well-documented exit codes.](https://github.com/resin-io/etcher/blob/master/lib/src/exit-codes.js)
+- A `--robot` option, which causes the Etcher CLI to output state in a way that can be easily machine-parsed.
+- An `--ipc` option, which causes the Etcher CLI to use [NodeJS's built-in IPC channel](https://nodejs.org/api/child_process.html#child_process_child_send_message_sendhandle_options_callback) to emit messages from the child to the parent.
+
+GUI fifty-thousand foot view
+----------------------------
+
+Given the event oriented nature of desktop application, it can be hard to follow what's going on without getting deep in the details.
+
+To mitigate this, we try to encapsulate functionality with nice and straightforward interfaces as AngularJS modules, and provide a single place where all the modules are tied together.
+
+Therefore, if you want to get a rough idea of how the GUI works, the perfect place to start is [application controller](https://github.com/resin-io/etcher/blob/master/lib/gui/app.js) and the [main view](https://github.com/resin-io/etcher/blob/master/lib/gui/partials/main.html), and diving into specific modules depending on your interests.
+
+Summary
+-------
+
+We always welcome contributions to our documentation. If you want to give back to Etcher, but feel that your knowledge on how it works is not enough to tackle a bug report or feature requests, use that as your advantage, since your fresh eyes could help unveil things that we take for granted, but should be documented instead!
```

Comment:
We always welcome contributions to our documentation -> We always welcome contributions to etcher as well as our documentation.